.github/workflows/reusable_provenance.yaml

```diff
@@ -0,0 +1,171 @@
+      # This only applies to `push` events, since it needs access to a valid Ent API key (which is
+      # not available on forks / PRs).
+      - name: Upload to Ent
+        if: github.event_name == 'push'
+        run: |
+          ent put ${{ steps.artifact_path.outputs.artifact_path }}
+
+      # Generate the input to the generate_provenance job.
+      # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+      - name: Generate artifact's base64-encoded hash
+        id: hash
+        run: |
+          # first cd to the directory containing the target binary. This allows us calling sha256
+          # on the target binary instead of having to provide the full path. This is fine for this
+          # experimental version, but has to be cleaned-up once we switch completely to SLSA
+          # provenance generation.
+          cd $(dirname ${{ steps.artifact_path.outputs.artifact_path }})
+          # sha256sum generates sha256 hash for the target binary.
+          # base64 --wrap=0 disables line wrapping, and encodes to base64 and outputs on a single line.
+          echo "hashes=$(sha256sum $(basename ${{ steps.artifact_path.outputs.artifact_path }}) | base64 --wrap=0)" >> $GITHUB_OUTPUT
+
+  # This job calls the generic SLSA workflow to generate provenance.
+  # See https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md
+  generate_provenance:
+    if: github.event_name == 'push'
+    needs: [build_binary]
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.1
+    with:
+      base64-subjects: '${{ needs.build_binary.outputs.hashes }}'
+      # Set a custom name for the provenance attestation.
+      attestation-name: '${{ needs.filename.outputs.filename }}.intoto.jsonl'
+      # Upload the provenance.
+      upload-assets: true
+      # Build the generator from source.
+      compile-generator: true
+
+  # This job uploads the signed provenance from the previous step to Ent, and
+  # publishes a comment about it on the PR.
+  upload_provenance:
+    if: github.event_name == 'push'
+    needs: [generate_provenance]
+    outputs:
+      provenance_digest: ${{ steps.ent_upload.outputs.provenance_digest }}
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Download the DSSE document
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ needs.filename.outputs.filename }}.intoto.jsonl
+
+      # See https://github.com/google/ent
+      - name: Download Ent CLI
+        if: steps.cache-ent.outputs.cache-hit != 'true'
+        env:
+          ENT_URL: https://ent-server-62sa4xcfia-ew.a.run.app
+          ENT_DIGEST: sha256:944a34854a2bf9c5d32f3bffa93885ee1c7ef8ab0f4fcc30898a981050ae4233
```

Comment:
Can this be shared with the other workflow too by creating a top level variable? I should probably create a reusable action for ent too at some point to encapsulate this, I think that would make things a bit cleaner, right?

Reply:
I think it cannot be shared, unless I make it an input to the reusable workflow. Do you think that is a good idea? If so, I can fix it in a future PR.